2025-04-03 00:56:29,769 - track_change - DEBUG - Created directory: /Users/samelder/sigfile/tracked_projects/sigfile/logs
2025-04-03 00:56:29,769 - track_change - DEBUG - Created directory: /Users/samelder/sigfile/tracked_projects/sigfile/changes
2025-04-03 00:56:29,769 - track_change - DEBUG - Created directory: /Users/samelder/sigfile/tracked_projects/sigfile/backups
2025-04-03 00:56:29,769 - track_change - DEBUG - Created directory: /Users/samelder/sigfile/tracked_projects/sigfile/ai_conversations
2025-04-03 00:56:29,769 - track_change - DEBUG - Created directory: /Users/samelder/sigfile/tracked_projects/sigfile/thinking
2025-04-03 00:56:29,769 - track_change - DEBUG - Created directory: /Users/samelder/sigfile/tracked_projects/sigfile/handoffs
2025-04-03 00:56:29,769 - track_change - INFO - Initializing OptimizedCapture for project: sigfile
2025-04-03 00:56:29,770 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,773 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,774 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,775 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,776 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,776 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,777 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,778 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,779 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,780 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,781 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,782 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,783 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,785 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,787 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,788 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,790 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,791 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,793 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,794 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,795 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,797 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,798 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,800 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,802 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,803 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,805 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,806 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,808 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,810 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,811 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,812 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,813 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,813 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,814 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,815 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,817 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,818 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,819 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,820 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,822 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,823 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,824 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,825 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,825 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,826 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,827 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,828 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,829 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,832 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,833 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,834 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,836 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,837 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,838 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,839 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,840 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,841 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,842 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,844 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,845 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,847 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,850 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,851 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,853 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,854 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,856 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,857 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,858 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,859 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,860 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,861 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,862 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,873 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,875 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,876 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,877 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,878 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,880 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,884 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,885 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,887 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,888 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,890 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,891 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,892 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,894 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,895 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,896 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,897 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,899 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,901 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,902 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,903 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,904 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,906 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,907 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,908 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,909 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,910 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,911 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,913 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,914 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,915 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,916 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,918 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,919 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,921 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,923 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,924 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,925 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,926 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,927 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,929 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,930 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,931 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,932 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,933 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,934 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,935 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,936 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,938 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,939 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,940 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,941 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,942 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,944 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,945 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,946 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,947 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,948 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,949 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,950 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,951 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,953 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,954 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,955 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,956 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,957 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,958 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,960 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,961 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,962 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,963 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,965 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,966 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,967 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,968 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,968 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,969 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,971 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,971 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,972 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,974 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,975 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,976 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,978 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,979 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,980 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,981 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,982 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,983 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,984 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,985 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,986 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,987 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,988 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,989 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,990 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,991 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,992 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,994 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,995 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,996 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,998 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:29,999 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,000 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,000 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,001 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,003 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,004 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,005 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,005 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,006 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,007 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,008 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,009 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,010 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,011 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,013 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,014 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,015 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,016 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,016 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,017 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,018 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,018 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,019 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,019 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,020 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,021 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,021 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,022 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,022 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,023 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,023 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,023 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,024 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,025 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,026 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,027 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,027 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,028 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,029 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,030 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,032 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,032 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,033 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,035 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,036 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,037 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,038 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,039 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,040 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,041 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,042 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,043 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,044 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,045 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,046 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,047 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,048 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,049 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,050 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,051 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,053 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,054 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,054 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,055 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,056 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,057 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,058 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,059 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,060 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,061 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,062 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,063 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,064 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,065 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,066 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,067 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,068 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,069 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,069 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,070 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,070 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,071 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,071 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,072 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,072 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,072 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,073 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,073 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,074 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,075 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,076 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,077 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,078 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,079 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,080 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,082 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,083 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,084 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,085 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,086 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,087 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,088 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,088 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,089 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,090 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,091 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,092 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,093 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,094 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,095 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,096 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,097 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,098 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,100 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,101 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,102 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,103 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,104 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,105 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,106 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,108 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,110 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,111 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,112 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,113 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,114 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,115 - track_change - ERROR - Error recording permission change: 'OptimizedCapture' object has no attribute 'file_roles'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 819, in _record_permission_change
    f.write(f"- Current Roles: {[r.value for r in self.file_roles.get(file_path, set())]}\n")
AttributeError: 'OptimizedCapture' object has no attribute 'file_roles'
2025-04-03 00:56:30,115 - track_change - INFO - Role-based permissions set up successfully
2025-04-03 00:56:30,115 - track_change - ERROR - Error setting up directories: [Errno 1] Operation not permitted: 'tracked_projects/sigfile/changes/.gitkeep'
Traceback (most recent call last):
  File "./src/scripts/track_change.py", line 557, in _setup_directories
    with open(gitkeep_path, 'w') as f:
PermissionError: [Errno 1] Operation not permitted: 'tracked_projects/sigfile/changes/.gitkeep'
